PSix/EventsListViewController.swift

```diff
@@ -7,15 +7,71 @@
 //
 
 import UIKit
+import Parse
 
-class EventsListViewController: UIViewController {
+class EventsListViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    static let EVENT_CELL_ID = "EventCell"
+    
+    private var userCreatedEvents = Events()
+    
+    @IBOutlet weak var eventsListTable: UITableView!
     
     override func viewWillAppear(animated: Bool) {
-        if !User.isLoggedIn() {
+        if !ParseUserSession.isLoggedIn {
             let onboardingStoryboard = UIStoryboard(name: "Onboarding", bundle: nil)
             let onboardingVC = onboardingStoryboard.instantiateViewControllerWithIdentifier("OnboardingViewController") as! UIViewController
             presentViewController(onboardingVC, animated: true, completion: nil)
+        } else if let currentUser = ParseUserSession.currentUser {
+            FacebookService.getFutureEventsCreatedByUser(currentUser, failure: nil) { [unowned self] (eventsData) -> Void in
+                self.extractEventsFromFbResponse(eventsData)
+                self.updateUI()
+            }
+        }
+        
+        eventsListTable.dataSource = self
+        eventsListTable.delegate = self
+    }
+    
+    func updateUI() {
+        eventsListTable.reloadData()
+    }
+    
+    func extractEventsFromFbResponse(events: NSDictionary) {
+        if let data = events["data"] as? NSArray {
+            for eventData in data {
+                if let fbId = eventData["id"] as? String,
+                   let ownerFbId = ParseUserSession.currentUser?["facebookId"] as? String,
+                   let name = eventData["name"] as? String {
+                    let event = Event(fbId: fbId, ownerFbId: ownerFbId, name: name)
+                    if let eventDescription = eventData["description"] as? String {
+                        event.eventDescription = eventDescription
+                    }
+                    if let location = eventData["location"] as? String {
+                        event.location = location
+                    }
+                    event.saveEventually()
+                    userCreatedEvents.append(event)
+                    
+                } else {
+                    println("Missing event information. Event object was not created.")
+                }
+            }
```

Comment:
This is a lot of logic that should sit in the VC.
Either have FacebookService return an array of events or have a third class handling this